test/python/unittest/API/APITest.py

```diff
@@ -0,0 +1,324 @@
+
+        self.assertTrue(hasattr(i.particles, "__getitem__"))
+        for ps in i.particles:
+            # self.assertIsInstance(ps, str)
+            self.assertIsInstance(i.particles[ps], openPMD.ParticleSpecies)
+
+    def testData_Order(self):
+        """ Test openPMD.Data_Order. """
+        obj = openPMD.Data_Order('C')
+        del obj
+
+    def testDatatype(self):
+        """ Test openPMD.Datatype. """
+        data_type = openPMD.Datatype(1)
+        del data_type
+
+    def testDataset(self):
+        """ Test openPMD.Dataset. """
+        data_type = openPMD.Datatype(1)
+        extent = [1, 1, 1]
+        obj = openPMD.Dataset(data_type, extent)
+        del obj
+
+    def testGeometry(self):
+        """ Test openPMD.Geometry. """
+        obj = openPMD.Geometry(0)
+        del obj
+
+    def testIteration(self):
+        """ Test openPMD.Iteration. """
+        self.assertRaises(TypeError, openPMD.Iteration)
+
+        iteration = self.__particle_series.iterations[400]
+
+        copy_iteration = openPMD.Iteration(iteration)
+
+        self.assertIsInstance(copy_iteration, openPMD.Iteration)
+
+    def testIteration_Container(self):
+        """ Test openPMD.Iteration_Container. """
+        obj = openPMD.Iteration_Container()
+        del obj
+
+    def testIteration_Encoding(self):
+        """ Test openPMD.Iteration_Encoding. """
+        obj = openPMD.Iteration_Encoding(1)
+        del obj
+
+    def testMesh(self):
+        """ Test openPMD.Mesh. """
+        self.assertRaises(TypeError, openPMD.Mesh)
+        mesh = self.__series.iterations[100].meshes['E']
+        copy_mesh = openPMD.Mesh(mesh)
+
+        self.assertIsInstance(copy_mesh, openPMD.Mesh)
+
+    def testMesh_Container(self):
+        """ Test openPMD.Mesh_Container. """
+        obj = openPMD.Mesh_Container()
+        del obj
+
+    def testParticlePatches(self):
+        """ Test openPMD.ParticlePatches. """
+        self.assertRaises(TypeError, openPMD.ParticlePatches)
+
+    def testParticleSpecies(self):
+        """ Test openPMD.ParticleSpecies. """
+        self.assertRaises(TypeError, openPMD.ParticleSpecies)
+
+    def testParticle_Container(self):
+        """ Test openPMD.Particle_Container. """
+        obj = openPMD.Particle_Container()
+        del obj
+
+    def testRecord(self):
+        """ Test openPMD.Record. """
+        # Has only copy constructor.
+        self.assertRaises(TypeError, openPMD.Record)
+
+        # Get a record.
+        electrons = self.__series.iterations[100].particles['electrons']
+        position = electrons['position']  # ['x']
+        self.assertIsInstance(position, openPMD.Record)
+
+        # Copy.
+        # copy_record = openPMD.Record(record)
+
+        # Check.
+        # self.assertIsInstance(copy_record, openPMD.Record)
+
+    def testRecord_Component(self):
+        """ Test openPMD.Record_Component. """
+        self.assertRaises(TypeError, openPMD.Record_Component)
+
+    def testFieldRecord(self):
```

Comment:
A ```Mesh``` is a ```Record``` by [design](https://github.com/openPMD/openPMD-standard/blob/1.1.0/STANDARD.md#mesh-based-records). What this actually tests are ```Mesh_Record_Component```s.